addon/-private.js

```diff
@@ -0,0 +1,5 @@
+export function recordDataFor(recordOrInternalModel) {
+  let internalModel = recordOrInternalModel._internalModel || recordOrInternalModel;
+
+  return internalModel._recordData || internalModel._modelData;
+}
```

Comment:
can we do this at build time with  https://github.com/ember-cli/ember-cli-version-checker  instead of adding this complexity at runtime?

if we have to do it at runtime, add comments to make it clear what code belongs to what version of ember data (3.4.x and 3.5.0 are the relevant versions i think)

Reply:
done